### PR TITLE
Created model of Book and repository with service for it using Spring…

### PR DESCRIPTION
**Pull request by Vlad1507**
… Boot.

### CODE REVIEW

File: src/main/java/com/store/bookstore/models/Book.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+@Entity
+@Data
+public class Book {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    private String title;
+    private String author;
+    private String isbn;
```
Comment (liliia-ponomarenko) [inline src/main/java/com/store/bookstore/models/Book.java:18]: Let's add appropriate requirements to the fields, for example for isbn
 ```
@Column(nullable = false, unique = true) 
 private String isbn;
```

File: src/main/java/com/store/bookstore/repository/impl/BookRepositoryImpl.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class BookRepositoryImpl implements BookRepository {
+    private final SessionFactory sessionFactory;
+
+    @Autowired
```
Comment (liliia-ponomarenko) [inline src/main/java/com/store/bookstore/repository/impl/BookRepositoryImpl.java:16]:
```suggestion
```
This annotation is optional

File: src/main/java/com/store/bookstore/repository/impl/BookRepositoryImpl.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+            session = sessionFactory.openSession();
+            transaction = session.beginTransaction();
+            session.persist(book);
+            transaction.commit();
+            return book;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new RuntimeException("Can't save book: " + book + " in the database");
```
Comment (liliia-ponomarenko) [inline src/main/java/com/store/bookstore/repository/impl/BookRepositoryImpl.java:35]:
```suggestion
            throw new DataProcessingException("Can't save book: " + book + " in the database");
```

File: src/main/java/com/store/bookstore/repository/impl/BookRepositoryImpl.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+            }
+        }
+    }
+
+    @Override
+    public List<Book> findAll() {
+        try (Session session = sessionFactory.openSession()) {
+            return session.createQuery("FROM Book", Book.class).getResultList();
+        } catch (Exception e) {
+            throw new RuntimeException("Can't get all books from database");
```
Comment (liliia-ponomarenko) [inline src/main/java/com/store/bookstore/repository/impl/BookRepositoryImpl.java:48]: the same

File: src/main/java/com/store/bookstore/services/impl/BookServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,20 @@
+package com.store.bookstore.services.impl;
+
+import com.store.bookstore.models.Book;
+import com.store.bookstore.repository.BookRepository;
+import com.store.bookstore.services.BookService;
+import java.util.List;
+
+public class BookServiceImpl implements BookService {
+    private BookRepository bookRepository;
```
Comment (liliia-ponomarenko) [inline src/main/java/com/store/bookstore/services/impl/BookServiceImpl.java:9]:
```suggestion
    private final BookRepository bookRepository;
```

**Review by okuzan — CHANGES_REQUESTED** [review]
git rm --cached .idea, same for target
they definitely shouldn't be in git
mvnw files are optional

File: src/main/java/com/store/bookstore/BookStoreApplication.java [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+    private BookService bookService;
+
+    public static void main(String[] args) {
+        SpringApplication.run(BookStoreApplication.class, args);
+    }
+
+    @Bean
+    public CommandLineRunner commandLineRunner() {
+        return args -> {
+            Book whiteFang = new Book();
```
Comment (Elena-Bruyako) [inline src/main/java/com/store/bookstore/BookStoreApplication.java:24]:
```suggestion
            Book book = new Book();
```

File: src/main/java/com/store/bookstore/exception/DataProcessingException.java [hunk 7]
```diff
@@ -0,0 +1,8 @@
+package com.store.bookstore.exception;
+
+public class DataProcessingException extends RuntimeException {
+
```
Comment (Elena-Bruyako) [inline src/main/java/com/store/bookstore/exception/DataProcessingException.java:4]:
```suggestion
```

remove empty line after class declaration

File: src/main/java/com/store/bookstore/models/Book.java [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import java.math.BigDecimal;
+import lombok.Data;
+
+@Entity
+@Data
```
Comment (Elena-Bruyako) [inline src/main/java/com/store/bookstore/models/Book.java:12]:
```suggestion
@Getter
@Setter
@Table(name...)
```

File: src/main/java/com/store/bookstore/repository/impl/BookRepositoryImpl.java [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+import com.store.bookstore.exception.DataProcessingException;
+import com.store.bookstore.models.Book;
+import com.store.bookstore.repository.BookRepository;
+import java.util.List;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.springframework.stereotype.Repository;
+
+@Repository
```
Comment (Elena-Bruyako) [inline src/main/java/com/store/bookstore/repository/impl/BookRepositoryImpl.java:12]:
```suggestion
@Repository
@RequiredArgsConstructor
```

File: src/main/java/com/store/bookstore/repository/impl/BookRepositoryImpl.java [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+public class BookRepositoryImpl implements BookRepository {
+    private final SessionFactory sessionFactory;
+
+    public BookRepositoryImpl(SessionFactory sessionFactory) {
+        this.sessionFactory = sessionFactory;
+    }
```
Comment (Elena-Bruyako) [inline src/main/java/com/store/bookstore/repository/impl/BookRepositoryImpl.java:18]:
```suggestion
```

File: src/main/java/com/store/bookstore/services/impl/BookServiceImpl.java [hunk 11]
```diff
@@ -0,0 +1,26 @@
+package com.store.bookstore.services.impl;
+
+import com.store.bookstore.models.Book;
+import com.store.bookstore.repository.BookRepository;
+import com.store.bookstore.services.BookService;
+import java.util.List;
+import org.springframework.stereotype.Service;
+
+@Service
```
Comment (Elena-Bruyako) [inline src/main/java/com/store/bookstore/services/impl/BookServiceImpl.java:9]:
```suggestion
@Service
@RequiredArgsConstructor
```

File: src/main/java/com/store/bookstore/services/impl/BookServiceImpl.java [hunk 12]
```diff
@@ -0,0 +1,26 @@
...
+public class BookServiceImpl implements BookService {
+    private final BookRepository bookRepository;
+
+    public BookServiceImpl(BookRepository bookRepository) {
+        this.bookRepository = bookRepository;
+    }
```
Comment (Elena-Bruyako) [inline src/main/java/com/store/bookstore/services/impl/BookServiceImpl.java:15]:
```suggestion
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments

**Review by Elena-Bruyako — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 13]
```diff
@@ -0,0 +1,18 @@
+name: Java CI
```
Comment (Elena-Bruyako) [inline .github/workflows/ci.yml:1]: add .idea to gitignore

File: src/main/java/com/store/bookstore/BookStoreApplication.java [hunk 14]
```diff
@@ -0,0 +1,16 @@
...
+
+import com.store.bookstore.services.BookService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+
+@SpringBootApplication
+public class BookStoreApplication {
+    @Autowired
+    private BookService bookService;
```
Comment (Elena-Bruyako) [inline src/main/java/com/store/bookstore/BookStoreApplication.java:11]: remove unused service

File: src/main/java/com/store/bookstore/dto/BookDto.java [hunk 15]
```diff
@@ -0,0 +1,16 @@
...
+@Data
+public class BookDto {
+    private Long id;
+    private String title;
+    private String author;
+    private String isbn;
+    private BigDecimal price;
+    private String description;
+    private String coverImage;
+
```
Comment (Elena-Bruyako) [inline src/main/java/com/store/bookstore/dto/BookDto.java:15]: remove empty line

```suggestion
```

File: src/main/java/com/store/bookstore/exception/DataProcessingException.java (hunk 7 above)
Comment (Elena-Bruyako) [reply src/main/java/com/store/bookstore/exception/DataProcessingException.java:4]: comment not fixed

File: src/main/java/com/store/bookstore/exception/EntityNotFoundException.java [hunk 16]
```diff
@@ -0,0 +1,8 @@
+package com.store.bookstore.exception;
+
+public class EntityNotFoundException extends RuntimeException {
+
```
Comment (Elena-Bruyako) [inline src/main/java/com/store/bookstore/exception/EntityNotFoundException.java:4]: remove empty line
```suggestion
```

File: src/main/java/com/store/bookstore/mapper/BookMapper.java [hunk 17]
```diff
@@ -0,0 +1,15 @@
...
+
+import com.store.bookstore.config.MapperConfig;
+import com.store.bookstore.dto.BookDto;
+import com.store.bookstore.dto.CreateBookRequestDto;
+import com.store.bookstore.models.Book;
+import org.mapstruct.Mapper;
+
+@Mapper(config = MapperConfig.class)
+public interface BookMapper {
+
```
Comment (Elena-Bruyako) [inline src/main/java/com/store/bookstore/mapper/BookMapper.java:11]: remove empty line

```suggestion
```

File: src/main/java/com/store/bookstore/models/Book.java (hunk 8 above)
Comment (Elena-Bruyako) [reply src/main/java/com/store/bookstore/models/Book.java:13]: comment not fixed

File: src/main/java/com/store/bookstore/repository/impl/BookRepositoryImpl.java [hunk 18]
```diff
@@ -0,0 +1,57 @@
...
+            throw new DataProcessingException("Can't get all books from database", e);
+        }
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        try (Session session = sessionFactory.openSession()) {
+            Book book = session.find(Book.class, id);
+            return Optional.ofNullable(book);
+        }
```
Comment (Elena-Bruyako) [inline src/main/java/com/store/bookstore/repository/impl/BookRepositoryImpl.java:55]: add catch block

File: src/main/java/com/store/bookstore/services/impl/BookServiceImpl.java [hunk 19]
```diff
@@ -0,0 +1,41 @@
...
+    @Override
+    public List<BookDto> findAll() {
+        return bookRepository.findAll().stream()
+                .map(bookMapper::toDto)
+                .toList();
+    }
+
+    @Override
+    public BookDto findById(Long id) {
+        return bookRepository.findById(id)
```
Comment (Elena-Bruyako) [inline src/main/java/com/store/bookstore/services/impl/BookServiceImpl.java:35]: stream works slowly so there is no purpose to use it
```suggestion
        Book book = bookRepository.findById(id).orElseThrow(
                        () -> new EntityNotFoundException("Can't get a book by id: " + id)
                );
        return bookMapper.toDto(book);
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments
Please, fix previous comments

File: pom.xml [hunk 20]
```diff
@@ -0,0 +1,155 @@
...
+	<repositories>
+		<repository>
+			<id>spring-milestones</id>
+			<name>Spring Milestones</name>
+			<url>https://repo.spring.io/milestone</url>
+			<snapshots>
+				<enabled>false</enabled>
+			</snapshots>
+		</repository>
+	</repositories>
```
Comment (liliia-ponomarenko) [inline pom.xml:143]: Why do you need it? And you duplicate it.

File: src/main/java/com/store/bookstore/dto/CreateBookRequestDto.java [hunk 21]
```diff
@@ -0,0 +1,16 @@
+package com.store.bookstore.dto;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import java.math.BigDecimal;
+import lombok.Data;
+
+@Data
+@JsonIgnoreProperties(ignoreUnknown = true)
```
Comment (liliia-ponomarenko) [inline src/main/java/com/store/bookstore/dto/CreateBookRequestDto.java:8]: Let’s  remove this annotation
```suggestion
```

**Review by liliia-ponomarenko — CHANGES_REQUESTED** [review]
Good job! But let’s fix my comment and issue with gitignore!

File: .github/workflows/ci.yml (hunk 13 above)
Comment (Elena-Bruyako) [reply .github/workflows/ci.yml:1]: I didin't find .gitignore file in your project, please add it

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
one comment

**Review by okuzan — APPROVED** [review]

**Pull request merged**